lib/unittest.bzl

```diff
@@ -22,7 +22,40 @@ assertions used to within tests.
 load(":new_sets.bzl", new_sets = "sets")
 load(":sets.bzl", "sets")
 
-_TOOLCHAIN_TYPE = "@bazel_skylib//toolchains:toolchain_type"
+TOOLCHAIN_TYPE = "@bazel_skylib//lib/unittest:toolchain_type"
+
+_UnittestToolchain = provider(
+    doc = "Execution platform information for rules in the bazel_skylib repository.",
+    fields = ["file_ext", "success_templ", "failure_templ", "join_on"],
+)
+
+def _unittest_toolchain_impl(ctx):
+    return [
+        platform_common.ToolchainInfo(
+            unittest_toolchain_info = _UnittestToolchain(
+                file_ext = ctx.attr.file_ext,
+                success_templ = ctx.attr.success_templ,
+                failure_templ = ctx.attr.failure_templ,
+                join_on = ctx.attr.join_on,
+            ),
+        ),
+    ]
+
+unittest_toolchain = rule(
+    implementation = _unittest_toolchain_impl,
+    attrs = {
+        "file_ext": attr.string(mandatory = True),
+        "success_templ": attr.string(mandatory = True),
+        "failure_templ": attr.string(mandatory = True),
+        "join_on": attr.string(mandatory = True),
+    },
+)
+
+def register_unittest_toolchains():
```

Comment:
I'm a bit uncomfortable mixing macros and rules intended for a BUILD file with those intended for a WORKSPACE file. Maybe I'm just over-sensitive? In either case, add a comment that this will only work if the macro is called from WORKSPACE.

Reply:
Done: added comment.
I can't think of any problems now about mixing workspace and build rules in the same .bzl file. If we later discover problems, it should be easy to split them.